### PR TITLE
Cover the HeaplessBigInt Ct surface + fix the Ct shift leak

### PR DESCRIPTION
**Pull request by kaidokert**
## What

Extends the constant-time asm-grep gate from `FixedUInt` to `HeaplessBigInt`, **fixes a real Ct shift leak** in the process, and closes the FixedUInt `Ord::cmp` gap. **378 fixtures, 0 violations on all 8 targets** (thumbv6m/7m/7em, riscv32imc/imac, aarch64, x86_64, avr-none); negative controls still trip.

## The library fix — heapless Ct shifts

`Shl`/`Shr<usize>`, `<u32>`, and `UnboundedShl`/`Shr` had **no `P::TAG` dispatch**, so `ct_value << secret` ran the branchful word/bit-shift and leaked the amount (FixedUInt routes its Ct arm through `const_shl_ct`). Fixed:

- Added heapless `const_shl_ct` / `const_shr_ct` — branchless barrel shifters (twins of FixedUInt's): each stage shifts by the public `2^k` via new width-preserving `shl_wp`/`shr_wp` and applies-or-not with a per-limb `black_box`-masked XOR select. Following FixedUInt (option B), the mask is `bit * T::MAX`, **not** subtle's `ConditionallySelectable` — so no bound cascade onto the P-generic operators. Also adds `ct_shr` (there was only `ct_shl`).
- All shift operators + unbounded now `match P::TAG`: Nct keeps the direct shift; Ct routes through the barrels. Ct `Shr` is width-preserving (a data-dependent `len` shrink would itself leak).

## The fixtures + allowlist

`fixtures_heapless.rs` covers the heapless `Ct` surface (saturating/abs_diff/bit-scan/is_zero, the subtle comparisons + conditional_select, the always-CT bitwise/wrapping/carry chains, `cmp`, the masked-return `Ct*` traits) **and now the shifts** (shl/shr/unbounded, is/next_power_of_two, midpoint). Built via a `heapless` cargo feature that `ct-driver` enables.

The heapless helpers pass because their per-limb loops bound on `len`/`max(len)`/`CAP` — all **public** — and each is **attested public-bounded** in `ct-driver/src/target.rs::HELPER_ALLOWLIST`, narrowly per-module, after reading each body. The dedicated shift barrels (`const_shl_ct`/`const_shr_ct`/`shl_wp`/`shr_wp`) are allowlisted like FixedUInt's; thumbv6m's `next_power_of_two` inherits the existing armv6m software-CLZ row. (An initial draft wrongly claimed this needed a `krabi-caliper` change — the allowlist is client-owned in-repo.)

FixedUInt `Ord::cmp` fixture added too (`const_cmp_ct` is now allowlisted — a stale `11` vs `12` mangled-length had been silently dropping it).

## is_one

Heapless `One::is_one` is now fixtured and gated too. Its Ct fold was already value-independent; the fixture-level violation was that it inlined into the fixture with its `len`-loop (which the strict fixture scan cannot allowlist), where `is_zero` is emitted out-of-line and is allowlisted. Fixed by extracting the fold to an `#[inline(never)]` `const_is_one_ct` helper (mirrors FixedUInt's, minus const-N unrolling) and attesting that symbol.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added a `heapless` feature for constant-time fixture generation, and enabled it in the verification driver.
  * Expanded constant-time fixture coverage for heapless big-integer operations (shifts, arithmetic, comparisons/equality, conditional selection, and bit utilities).

* **Bug Fixes**
  * Improved constant-time shift and carry behavior to avoid data-dependent control flow.
  * Fixed constant-time equality (`FixedUInt` Ct) and “is one” handling for canonical/edge cases.

* **Tests**
  * Added/expanded constant-time correctness tests for equality, “is one,” and shift matching between Ct and non-Ct personalities.

### CODE REVIEW

File: ct-verify/ct-fixtures/src/lib.rs [hunk 1]
```diff
@@ -28,6 +28,8 @@ mod fixtures_cat_a;
 mod fixtures_cat_b;
 mod fixtures_cat_c;
 mod fixtures_ct_traits;
+#[cfg(feature = "heapless")]
+mod fixtures_heapless;
```
[the remark this answers is not in the document]
Comment (kaidokert) [reply ct-verify/ct-fixtures/src/lib.rs:32]: Resolved in 98f087b, the commit right after the scaffolding one reviewed here (0e2aa1d). `fixture_features` is now `&["heapless"]`, so `cargo run -p ct-driver` builds ct-fixtures with the feature and scans the `ct_fix__H*` symbols — 393 fixtures across all 8 targets, up from the FixedUInt-only set. No separate `--features` forwarding is needed since the driver sets it in `DriverConfig`.
